aui.views/src/AUI/View/AViewContainer.cpp

```diff
@@ -40,31 +50,64 @@ void AViewContainer::drawView(const _<AView>& view, ClipOptimizationContext cont
         return;
     }
 
-    if (view->mSkipUntilLayoutUpdate) [[unlikely]] {
-        view->mSkipUntilLayoutUpdate = false;
-        return;
-    }
+    const auto prevStencilLevel = contextOfTheView.render.getStencilDepth();
 
-    const auto prevStencilLevel = ARender::getRenderer()->getStencilDepth();
+    const bool showRedraw = [&] {
+        if (view->mRedrawRequested) [[unlikely]] {
+            if (auto w = AWindow::current()) [[unlikely]] {
+                if (w->profiling().displayRedrawRequests) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }();
+    AUI_DEFER {
+        if (showRedraw) [[unlikely]] {
+            auto c = contextOfTheView.render.getColor();
+            AUI_DEFER { contextOfTheView.render.setColorForced(c); };
+            contextOfTheView.render.rectangle(ASolidBrush{0x80ff00ff_argb}, view->getPosition(), view->getSize());
+        }
+    };
 
-    RenderHints::PushState s;
-    glm::mat4 t(1.f);
-    view->getTransform(t);
-    ARender::setColor(AColor(1, 1, 1, view->getOpacity()));
-    ARender::setTransform(t);
+    RenderHints::PushState s(contextOfTheView.render);
+    auto applyTransform = [&] {
+        glm::mat4 t(1.f);
+        view->getTransform(t);
+        contextOfTheView.render.setTransform(t);
+        contextOfTheView.render.setColor(AColor(1, 1, 1, view->getOpacity()));
+    };
+    if (view->mRenderToTexture) [[unlikely]] {
+        auto invalidAreas = std::exchange(view->mRenderToTexture->invalidAreas, {});
+        if (invalidAreas.empty()) {
+//            goto drawRenderToTexture;
```

Comment:
Commnted code should be removed. In general, goto is a bad practice and it's easy to avoid its usage here via lambda functions.

Reply:
Deprecated